### ISSUE TITLE
Bug: [consistent-type-imports] erroneous report on type used in an `export =` assertion

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#8330]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

n/a - the broken commit hasn't released yet so it doesn't repro on playground

### Repro Code

```TypeScript
import type Foo from 'foo';

export = {} as Foo;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-imports": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error

### Actual Result

Type import "Foo" is used by decorator metadata

### Additional Info

Looks like it was introduced by #8265
